cmd/oras/internal/option/packer.go

```diff
@@ -115,11 +115,20 @@ func (opts *Packer) parseAnnotations(cmd *cobra.Command) error {
 }
 
 // decodeJSON decodes a json file v to filename.
-func decodeJSON(filename string, v interface{}) error {
+func decodeJSON(filename string, v interface{}) (err error) {
 	file, err := os.Open(filename)
 	if err != nil {
 		return err
 	}
-	defer file.Close()
-	return json.NewDecoder(file).Decode(v)
+	defer func() {
+		closeErr := file.Close()
+		if err == nil {
+			err = closeErr
+		}
+	}()
+	err = json.NewDecoder(file).Decode(v)
+	if err != nil {
+		return err
+	}
+	return err
 }
```

Comment:
I was wondering if I was going to be dinged for coverage on this private method. Difficult thing to test and little value of testing it.